core/src/main/scripts/importer/validateData.py

```diff
@@ -1484,16 +1485,22 @@ def checkLine(self, data):
         if 'Hugo_Symbol' in self.cols:
             hugo_symbol = data[self.cols.index('Hugo_Symbol')].strip()
             # treat the empty string or 'Unknown' as a missing value
-            if hugo_symbol in ('', 'Unknown'):
+            if hugo_symbol in ('NA', '', 'Unknown'):
                 hugo_symbol = None
         if 'Entrez_Gene_Id' in self.cols:
             entrez_id = data[self.cols.index('Entrez_Gene_Id')].strip()
             # treat the empty string or 0 as a missing value
-            if entrez_id in ('', '0'):
+            if entrez_id in ('NA', '', '0'):
                 entrez_id = None
         # validate hugo and entrez together:
-        self.checkGeneIdentification(hugo_symbol, entrez_id)
-
+        normalized_gene = self.checkGeneIdentification(hugo_symbol, entrez_id)
+        
+        # validate the gene to make sure its from the targeted panel
+        if normalized_gene and self.portal.gene_panel_list and data[sample_id_column_index] in sample_ids_panel_dict:
+            panel_id = sample_ids_panel_dict[data[sample_id_column_index]]
+            if panel_id in self.portal.gene_panel_list and panel_id != 'NA':
```

Comment:
Do we still need to check `NA` at here? Not sure will gene_panel_list contains `NA`.

Reply:
Thanks for the review @dippindots. The panel id sometimes is `NA`. For example see: https://github.com/cBioPortal/cbioportal/blob/master/core/src/test/scripts/test_data/study_es_0/data_gene_panel_matrix.txt